src/terms/pairwise.jl

```diff
@@ -1,3 +1,9 @@
+# We cannot use `LinearAlgebra.norm` with complex numbers due to the need to use its
+# analytic continuation
```

Comment:
Comment not great. "Complex-analytic extension of norm(x) from real to complex inputs."

Reply:
Also explain a bit the reasoning behind this. Eg "We want to perform a matrix-vector product f'(x) * h, where f is a real-to-real function, and h a complex vector. To do this using automatic differentiation, we can extend analytically f to accept complex inputs, then differentiate t -> f(x+t h). This will fail if non-analytic functions like `norm` are used for complex inputs, and therefore we have to redefine it."

Reply:
This will probably be needed for other functions, so longer-term it should live in common/, but I would say fine for now.